flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/DefaultExecutionGraph.java

```diff
@@ -1276,6 +1276,14 @@ private void maybeReleasePartitions(final Execution attempt) {
 
     private void releasePartitions(final List<IntermediateResultPartitionID> releasablePartitions) {
         if (releasablePartitions.size() > 0) {
+
+            // Remove cached ShuffleDescriptor when partition is released
```

Comment:
Would any problem happen without this newly added line but only notify in `IntermediateResultPartition#resetForNewExecution()`?

Reply:
`resetForNewExecution()` removes the cache when failover occurs. `releasePartitions` removes the cache when the partition is released. This will make sure when partitions are no longer valid, the cache related to the partitions will be removed, too.

Reply:
Currently when a finished partition is released due to no longer needed, it is still consumable. So I'm thinking whether it's really needed to clear the cache.